util/topgen/secure_prng.py

```diff
@@ -0,0 +1,289 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+"""Generating random variables by using a cryptographically secure PRNG.
+
+A class for generating random numbers and permutations.
+This class is based on a cryptographically secure PRNG CTR_DRBG described in
+NIST SP 800-90A, section 10.2.1
+https://csrc.nist.gov/publications/detail/sp/800-90a/rev-1/final
+This PRNG guarantees forward and backward secrecy (PRNG output cannot be
+computed from any previous or future outpus) as well as enhanced backward
+secrecy (previous PRNG outputs cannot be computed from a compromised
+internal state).
+"""
+
+import logging as log
+import sys
+from Crypto.Cipher import AES
+from math import ceil as _ceil
+from math import log as _log
+
+
+class secure_prng():
+
+    blocklen = 128
+    keylen = 128
+    # seedlen = blocklen + keylen = 256
+    # ctr_len = blocklen = 128
```

Comment:
We should probably remove this commented out code.

Reply:
FYI: there have been previous discussions on this before `seedlen` and `ctr_len` are the terms used in the NIST spec. Due to the simplified / stripped-down implementation we've selected here, they are not used in the code, but @vrozic 's code comments refer to them. That's why it was decided to keep the variables but comment them. @vrozic it probably makes sense to add a comment on this here. The same question might come up again. This is already the second time ;-)

Reply:
> We should probably remove this commented out code.

I've added some text to explain it.